src/secw_security_wallet_server.cc

```diff
@@ -36,684 +36,809 @@
 
-    m_supportedCommands[GET_WITHOUT_SECRET_BY_NAME] = std::bind(&SecurityWalletServer::handleGetDocumentWithoutSecretByName, this, _1, _2);
-    m_supportedCommands[GET_WITH_SECRET_BY_NAME] = std::bind(&SecurityWalletServer::handleGetDocumentWithSecretByName, this, _1, _2);
-    
-    m_supportedCommands[CREATE] = std::bind(&SecurityWalletServer::handleCreate, this, _1, _2);
-    m_supportedCommands[DELETE] = std::bind(&SecurityWalletServer::handleDelete, this, _1, _2);
-    m_supportedCommands[UPDATE] = std::bind(&SecurityWalletServer::handleUpdate, this, _1, _2);
-}
+        m_supportedCommands[GET_WITHOUT_SECRET] = std::bind(&SecurityWalletServer::handleGetDocumentWithoutSecret, this, _1, _2);
+        m_supportedCommands[GET_WITH_SECRET] = std::bind(&SecurityWalletServer::handleGetDocumentWithSecret, this, _1, _2);
 
-std::vector<std::string> SecurityWalletServer::handleRequest(const Sender & sender, const std::vector<std::string> & payload)
-{
-    try
+        m_supportedCommands[GET_WITHOUT_SECRET_BY_NAME] = std::bind(&SecurityWalletServer::handleGetDocumentWithoutSecretByName, this, _1, _2);
+        m_supportedCommands[GET_WITH_SECRET_BY_NAME] = std::bind(&SecurityWalletServer::handleGetDocumentWithSecretByName, this, _1, _2);
+
+        m_supportedCommands[CREATE] = std::bind(&SecurityWalletServer::handleCreate, this, _1, _2);
+        m_supportedCommands[DELETE] = std::bind(&SecurityWalletServer::handleDelete, this, _1, _2);
+        m_supportedCommands[UPDATE] = std::bind(&SecurityWalletServer::handleUpdate, this, _1, _2);
+    }
+
+    std::vector<std::string> SecurityWalletServer::handleRequest(const Sender & sender, const std::vector<std::string> & payload)
     {
-        if(payload.size() == 0)
+        try
         {
-            throw SecwProtocolErrorException("Command frame is empty");
+            if(payload.size() == 0)
+            {
+                throw SecwProtocolErrorException("Command frame is empty");
+            }
+
+            Command cmd = payload.at(0);
+
+            if(cmd == "ERROR" || cmd == "OK")
+            {
+                //avoid loop
+                return {};
+            }
+
+            //check if the command exist in the system
+            if( m_supportedCommands.count(cmd) == 0)
+            {
+                throw SecwUnsupportedCommandException(cmd + " not supported");
+            }
+
+            FctCommandHandler cmdHandler = m_supportedCommands[cmd];
+
+            // Declaring new vector 
+            std::vector<std::string> params(payload.begin()+1, payload.end());
+
+            std::string result = cmdHandler(sender, params);
+
+            return {result};
         }
-        
-        Command cmd = payload.at(0);
-        
-        if(cmd == "ERROR" || cmd == "OK")
+        catch(SecwException &e)
         {
-            //avoid loop
-            return {};
+            log_warning("%s", e.what());
+            return {"ERROR", e.toJson()};
```

Comment:
I'm not sure how this will influence JSONIFY(...) macro in fty-security-wallet-rest. Won't it need an update to keep working the same?